packages/amplify-prompts/src/progressbars/progressbar.ts

```diff
@@ -0,0 +1,252 @@
+import { AmplifyTerminal as Terminal } from './terminal';
+
+/**
+ * Type for Bar configurations options
+ */
+export type BarOptions = {
+    // Custom formatter functions for main progress bar and item's underneath it
+    progressBarFormatter: (payload: ProgressPayload) => string,
+    itemFormatter: (payload: ItemPayload) => string,
+    // Indicates if the progress bar is part of a multiBar or standalone
+    loneWolf: boolean,
+    hideCursor: boolean,
+    // Specify characters for a completed/incomplete bar
+    barCompleteChar: string,
+    barIncompleteChar : string,
+    barSize: number,
+    // status which indicates item completion/failure
+    itemCompleteStatus: string[],
+    itemFailedStatus: string[],
+    // Text which appear on top of the progress bar indicating current status along with a spinner.
+    prefixText: string,
+    successText: string,
+    failureText: string
+}
+
+/**
+ * Type for main progress bar payload
+ */
+export type ProgressPayload = {
+    progressName: string,
+    envName: string
+}
+
+/**
+ * Type for progress bar item payload
+ */
+export type ItemPayload = {
+    LogicalResourceId: string,
+    ResourceType: string,
+    ResourceStatus: string,
+    TimeStamp: string,
+}
+
+type Item = {
+    name: string,
+    status: string,
+    renderString: string,
+    finished: boolean
+}
+
+/**
+ * Progress bar instance
+ */
+export class ProgressBar {
```

Comment:
did we investigate using [`progress`](https://www.npmjs.com/package/progress) for this? It's a lightweight package that we are already using in a few other places to display a progress bar. Not sure how well it will work for this use case since we are displaying multiple bars with other status in between, just want to make sure we aren't reinventing the wheel